### ISSUE TITLE
ipfs.object.put()'s CID is not a valid CID object

### ISSUE THREAD

**Issue by koalalorenzo** [ipfs/js-ipfs-http-client#904]
I am not sure if this is a bug or something  but according to the documentation I should be able to get the CID correctly:

```
ipfs.object.put(myDAG).then((cid) => console.log(cid.toString()))
```

The docs I am referring to: https://github.com/ipfs/interface-ipfs-core/blob/master/SPEC/OBJECT.md#objectput
CIDs docs: https://github.com/multiformats/js-cid

It seems that one of the two docs are wrong about `.toString()`.
Any Idea how to get the multihash? I was not lucky by trying several things, including creating a new CID from the values that I got. Any idea?

Once I know I could make a PR to update the docs in the other repo.

**Comment by koalalorenzo** [comment]
I am not able to reproduce the bug since I changed a lot of code since then (sorry).
It was not returning any error, as it would have been caught in the promise chain (there was a .catch).
I tried to reproduce it manually from Node, and it seems ok (toString is fine).

If it helps, this happened inside a web worker on Safari, Chrome and Firefox: the promise returned something that looked like a "serialised" CID object with explained why `.toString()` was returning `[object object]` 😄 We wrote the whole web worker in Typescript + Webpack. That might be one of the reason but not sure.

I made a workaround as I was manipulating something simple to wrap custom objects just added. But it could be a simple issue in the Web Worker with CID. I see that NodeJS is fine with `.toString()`
